src/main/VcIssuanceClient.ts

```diff
@@ -71,20 +73,46 @@ export class VcIssuanceClient {
    * @param opts
    *         - jwtSignerCallback: function to sign the proof
    *         - jwtSignerArgs: The arguments to create the signature
+   *         - jwtVerifyCallback: function to verify if JWT is valid
    */
   public async createProofOfPossession(opts: {
     jwtSignerCallback: JWTSignerCallback;
     jwtSignerArgs: JWTSignerArgs;
+    jwtVerifyCallback: JWTVerifyCallback;
   }): Promise<ProofOfPossession> {
-    if (!opts.jwtSignerCallback || !opts.jwtSignerArgs) {
+    if (!opts.jwtSignerCallback || !opts.jwtSignerArgs || !opts.jwtVerifyCallback) {
       throw new Error(BAD_PARAMS);
     }
+    const signerArgs = this.setJWSDefaults(opts.jwtSignerArgs);
+    const jwt = await opts.jwtSignerCallback(signerArgs);
+    try {
+      const algorithm = opts.jwtSignerArgs.header.alg;
```

Comment:
A simple check whether it started with 'ey' contained 2 dots would have been enough, but this is nicer. Although this does force the implementor to provide a jwtVerifyCallback implementation.

Maybe make that optional. If present use the verify callback to verify the JWT. IF not present use the simple version checking for 3 parts seperated by dots and starting with ey

Reply:
Fixed!